crates/sui-types/src/sui_system_state/mod.rs

```diff
@@ -378,6 +382,29 @@ where
     }
 }
 
+pub fn get_validators_from_table_vec<S, ValidatorType>(
+    object_store: &S,
+    table_id: ObjectID,
+    table_size: u64,
+) -> Result<Vec<ValidatorType>, SuiError>
+where
+    S: ObjectStore,
+    ValidatorType: Serialize + DeserializeOwned,
+{
+    let mut validators = vec![];
+    for i in 0..table_size {
+        let validator: ValidatorType = get_dynamic_field_from_store(object_store, table_id, &i)
+            .map_err(|err| {
+                SuiError::SuiSystemStateReadError(format!(
+                    "Failed to load validator from table: {:?}",
+                    err
+                ))
+            })?;
+        validators.push(validator);
```

Comment:
I think we could use an iterator here to avoid multiple point look up. Maybe a TODO item